hydra_base/lib/scenario.py

```diff
@@ -1377,9 +1412,12 @@ def get_resource_data(ref_key,
 
     user_id = kwargs.get('user_id')
 
-    resource_i = get_resource(ref_key, ref_id)
-
-    resource_attributes = resource_i.attributes
+    if ref_key is not None and ref_id is not None:
```

Comment:
If there was only a `ref_key` we could return attributes for only the matching `ref_key` type, i.e. instead of falling back to `_get_all_network_resource_attributes` use something like `_get_all_<<ref_key_type>>_resource_attributes`.